apis/logging/v1/cluster_log_forwarder.go

```diff
@@ -33,13 +33,17 @@ func IsOutputTypeName(s string) bool {
 }
 
 // Get all subordinate condition messages for condition of type "Ready" and False
+// A 'true' Ready condition with a message means some error with pipeline but it is still valid
 func (status ClusterLogForwarderStatus) GetReadyConditionMessages() []string {
 	var messages = []string{}
 	for _, nc := range []NamedConditions{status.Pipelines, status.Inputs, status.Outputs} {
 		for _, conds := range nc {
+			currCond := conds.GetCondition(ConditionReady)
 			if !conds.IsTrueFor(ConditionReady) {
-				currCond := conds.GetCondition(ConditionReady)
 				messages = append(messages, currCond.Message)
+				// If a pipeline is "degraded" then it should have an extra error message
```

Comment:
In general, the existence of the message should be immaterial to the whether the condition applies or not.

Reply:
N/A. "Degraded" piplines are invalid/ "degraded" is not removed.